packages/@o3r/core/schematics/component/common/theming.ts

```diff
@@ -0,0 +1,56 @@
+import { externalSchematic, Rule, SchematicContext } from '@angular-devkit/schematics';
+import { askConfirmation } from '@angular/cli/src/utilities/prompt';
+import { setupSchematicsDefaultParams } from '@o3r/schematics';
+import { NgGenerateComponentSchematicsSchema } from '../schema';
+
+const stylingPackageName = '@o3r/styling';
+const addThemingSchematicName = 'theming-to-component';
+
+export const getAddThemingRules = async (
+  stylePath: string,
+  options: Pick<NgGenerateComponentSchematicsSchema, 'useOtterTheming'>,
+  context: SchematicContext
+): Promise<Rule[]> => {
+  let useOtterTheming = options.useOtterTheming;
+  let alwaysUseOtterTheming = false;
```

Comment:
Instead of boolean I would use a `boolean | undefined`.
The purpose is that to ask the question only once by modifying the next part of the code with:

```typescript
  return useOtterTheming ? [
    externalSchematic(stylingPackageName, addThemingSchematicName, {
      path: stylePath
    }),
    ...(alwaysUseOtterTheming !== undefined ? [
      setupSchematicsDefaultParams({
        // eslint-disable-next-line @typescript-eslint/naming-convention
        '@o3r/core:component': {
          useOtterTheming: alwaysUseOtterTheming
        },
        // eslint-disable-next-line @typescript-eslint/naming-convention
        '@o3r/core:component-presenter': {
          useOtterTheming: alwaysUseOtterTheming
        }
      })
    ] : [])
  ] : [];
```